src/twisted/test/test_defgen.py

```diff
@@ -338,6 +339,58 @@ def _noYield():
 
         self.assertIn("inlineCallbacks", str(self.assertRaises(TypeError, _noYield)))
 
+    def test_internalDefGenReturnDoesntLeak(self):
+        """
+        When one inlineCallbacks calls another, the internal L{_DefGen_Return}
+        flow control exception doesn't leak into tracebacks captured in the
+        caller.
+        """
+        clock = task.Clock()
+
+        @inlineCallbacks
+        def _returns():
+            yield task.deferLater(clock, 0)
+            returnValue(6)
+
+        @inlineCallbacks
+        def _raises():
+            try:
+                yield _returns()
+                raise TerminalException("boom")
```

Comment:
Very minor comment to make the exaction unique and make it easier to trace the origin of this exception, in the case in which things go very bad and we have random side effects

```suggestion
                raise TerminalException("boom returnValue")
```